### PR TITLE
Fix 'React is used before defined' in TS files

### PR DESCRIPTION
**Pull request by CompuIves**
For TS files there needs to be a different check, based on https://stackoverflow.com/questions/63818415/react-was-used-before-it-was-defined.

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5000/commits/d0c534a44ca040b36693a444e901a7e2a563ecbe is at https://pr5000.build.csb.dev/s/new.

File: packages/app/src/app/overmind/effects/vscode/LinterWorker/index.js [hunk 1]
```diff
@@ -371,6 +373,9 @@ const TYPESCRIPT_PARSER_OPTIONS = {
     // Fixes optional chaining
     'no-unused-expressions': 'off',
     '@typescript-eslint/no-unused-expressions': 'warn',
+    // note you must disable the base rule as it can report incorrect errors
+    'no-use-before-define': 'off',
+    '@typescript-eslint/no-use-before-define': ['error'],
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/effects/vscode/LinterWorker/index.js:378]:
```suggestion
    '@typescript-eslint/no-use-before-define': 'error',
```

**Review by christianalfoni — CHANGES_REQUESTED** [review]
Just for consistency 😄

**Comment by CompuIves** [comment]
Ah good one!

**Review by christianalfoni — APPROVED** [review]
Great! 👍

**Pull request merged**